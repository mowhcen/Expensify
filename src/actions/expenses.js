import database from "../firebase/firebase";

// ADD_EXPENSE
const addExpense = (expense) => ({
    type: "ADD_EXPENSE",
    expense,
});

const startAddExpense = (expenseData = {}) => {
    return (dispatch) => {
        const {
            description = "",
            note = "",
            amount = 0,
            createAt = 0,
        } = expenseData;

        const expense = { description, note, amount, createAt };

        return database
            .ref("expenses")
            .push(expense)
            .then((ref) => {
                dispatch(
                    addExpense({
                        id: ref.key,
                        ...expense,
                    })
                );
            });
    };
};

// REMOVE_EXPENSE
const removeExpense = ({ id } = {}) => ({
    type: "REMOVE_EXPENSE",
    id,
});
// EDIT_EXPENSE
const editExpense = (id, updates) => ({
    type: "EDIT_EXPENSE",
    id,
    updates,
});

// SET_EXPENSES
const setExpenses = (expenses) => ({
    type: "SET_EXPENSES",
    expenses,
});

const startSetExpenses = () => {
    return (dispatch) => {
        return database
            .ref("expenses")
            .once("value")
            .then((snapshot) => {
                const expenses = [];

                snapshot.forEach((childSnapshot) => {
                    expenses.push({
                        id: childSnapshot.key,
                        ...childSnapshot.val(),
                    });
                });

                dispatch(setExpenses(expenses));
            });
    };
};

export {
    removeExpense,
    editExpense,
    addExpense,
    startAddExpense,
    setExpenses,
    startSetExpenses,
};
