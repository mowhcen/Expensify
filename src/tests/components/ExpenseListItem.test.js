import ExpenseListItem from "../../components/ExpenseListItem";
import React from "react";
import expenses from "../fixtures/expenses";
import { shallow } from "enzyme";

test("should render ExpenseListItem with a expense", () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[2]} />);
    expect(wrapper).toMatchSnapshot();
});
