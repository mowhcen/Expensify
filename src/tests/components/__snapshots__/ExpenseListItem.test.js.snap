// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with a expense 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/3"
  >
    <h3>
      Credit Card
    </h3>
  </Link>
  <p>
    $45.67
    -
    January 5th, 1970
  </p>
</div>
`;
